src/core/Form/SearchInput/SearchInput.test.tsx

```diff
@@ -199,6 +199,45 @@ describe('props', () => {
       }
     });
   });
+  describe('debounce', () => {
+    it('delays the running of onChange by the given time', () => {
+      jest.useFakeTimers();
+      const mockOnChange = jest.fn();
+      const searchInput = (
+        <SearchInput
+          labelText="Debounced search"
+          clearButtonLabel="clear"
+          searchButtonLabel="search"
+          debounce={1000}
+          onChange={mockOnChange}
+        />
+      );
+      const { getByRole } = render(searchInput);
+
+      const inputElement = getByRole('searchbox') as HTMLInputElement;
+      fireEvent.change(inputElement, { target: { value: 'new value' } });
+      expect(mockOnChange).not.toBeCalled();
+      jest.advanceTimersByTime(1000);
+      expect(mockOnChange).toBeCalledTimes(1);
+      expect(inputElement.value).toBe('new value');
+    });
+    it('resolves right when no onChange is given', () => {
```

Comment:
I'd rather test that it works as expected when no debounce is given, would be more common scenario.

Reply:
Ok, seems like there is already such a test written before this issue, so no need to duplicate it. Renaming this test would still be desirable in my opinion.

Reply:
Removed it from here and added a similar one to Debounce's own tests.